nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/encryption/EncryptedS3ClientService.java

```diff
@@ -0,0 +1,297 @@
+                    new AllowableValue(CryptoMode.AuthenticatedEncryption.toString(), "Authenticated Encryption"),
+                    new AllowableValue(CryptoMode.EncryptionOnly.toString(), "Encryption Only"))
+            .description("Cryptographic mode used to secure an S3 object.")
+            .build();
+
+    public static final PropertyDescriptor CRYPTO_STORAGE_MODE = new PropertyDescriptor.Builder()
+            .name("storage-mode")
+            .displayName("Crypto Storage Mode")
+            .required(false)
+            .allowableValues(
+                    new AllowableValue(CryptoStorageMode.InstructionFile.toString(), "Instruction File"),
+                    new AllowableValue(CryptoStorageMode.ObjectMetadata.toString(), "Object Metadata"))
+            .description("Storage mode used to store encryption information when encrypting an S3 object.")
+            .build();
+
+    public static final PropertyDescriptor IGNORE_MISSING_INSTRUCTION_FILE = new PropertyDescriptor.Builder()
+            .name("ignore-missing-instruction-file")
+            .displayName("Ignore Missing Instruction File")
+            .required(false)
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .description("Whether to ignore missing instruction files during decryption.")
+            .build();
+
+    private static final List<PropertyDescriptor> properties;
+    private String encryptionMethod;
+    private String kmsCmkId;
+    private String kmsRegion;
+    private String secretKey;
+    private String secretKeyAlgorithm;
+    private String privateKey;
+    private String privateKeyAlgorithm;
+    private String publicKey;
+    private String publicKeyAlgorithm;
+    private String cryptoMode;
+    private String cryptoStorageMode;
+    private Boolean ignoreMissingInstructionFile;
+
+
+    static {
+        final List<PropertyDescriptor> props = new ArrayList<>();
+        props.add(ENCRYPTION_METHOD);
+        props.add(KMS_CMK_ID);
+        props.add(KMS_REGION);
+        props.add(SECRET_KEY);
+        props.add(SECRET_KEY_ALGORITHM);
+        props.add(PRIVATE_KEY);
+        props.add(PRIVATE_KEY_ALGORITHM);
+        props.add(PUBLIC_KEY);
+        props.add(PUBLIC_KEY_ALGORITHM);
+        props.add(CRYPTO_MODE);
+        props.add(CRYPTO_STORAGE_MODE);
+        props.add(IGNORE_MISSING_INSTRUCTION_FILE);
+
+        properties = Collections.unmodifiableList(props);
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @OnEnabled
+    public void onConfigured(final ConfigurationContext context) throws InitializationException {
+        encryptionMethod = context.getProperty(ENCRYPTION_METHOD).getValue();
+        kmsCmkId = context.getProperty(KMS_CMK_ID).getValue();
```

Comment:
Should this evaluate expression language like `context.getProperty(KMS_CMK_ID).evaluateAttributeExpressions().getValue()`?